src/core/ext/client_channel/client_channel.c

```diff
@@ -299,6 +302,10 @@ static void on_resolver_result_changed(grpc_exec_ctx *exec_ctx, void *arg,
   }
 
   gpr_mu_lock(&chand->mu);
+  if (lb_policy_name != NULL) {
+    gpr_free(chand->lb_policy_name);
+    chand->lb_policy_name = lb_policy_name;
```

Comment:
Related to a previous comment, having `chand->lb_policy_name = gpr_strdup(lb_policy_name)` would replace all the new `gpr_strdup`s and save the free on line 270.